### DISCUSSION TITLE
Realtime API?

### DISCUSSION THREAD

**Discussion by Hugobox** [allanbunch/node-red-openai-api#28] [Ideas]
Hi,

I love your Node-RED OpenAI implementation! The speech generation feature works great. I'd like to request an update to the OpenAI-node dependency to v4.69.0, which enables the Realtime API. This would unlock new possibilities for my projects.

Could you consider upgrading the dependency to take advantage of this new feature?

**Comment by allanbunch** [comment]
You bet! Great to hear that you're finding the node useful. Look for the dependency version bump later today.

**Reply by allanbunch** [reply]
@Hugobox We're now running with the latest OpenAI Node API library; [v4.72.0](https://www.npmjs.com/package/openai/v/4.72.0). You can find my release notes  couple of release notes [here ](https://github.com/allanbunch/node-red-openai-api/tree/v1.6.0?tab=readme-ov-file#release-notes-v160). Enjoy!

**Comment by Den-Dennis** [comment]
@allanbunch Also from me, thanks for creating and maintaining this. Love how you did this.
@Hugobox can i ask you to share a test flow if you get the Realtime API working?

**Reply by allanbunch** [reply]
Thanks. Great to hear.

The [Realtime API](https://platform.openai.com/docs/api-reference/realtime) is pretty sparse at the moment. I've found that quite a bit of code is necessary to make something useful. Here's an early [example flow](https://flows.nodered.org/flow/cb0123996b8ccbc8f026ebe780bfd76d) I did as I was learning the Realtime API. I've since refined my approach--e.g. removed the MQTT node requirement--though my latest still requires a bunch of Dashboard Template node code to handle all the audio buffering, etc. I suspect that OpenAI is working to expand the Realtime implementation as we speak.
